gdtf.py

```diff
@@ -531,23 +532,11 @@ def add_child_position(geometry):
 
             # if (not sanitize_obj_name(geometry) in objs): return
             obj_child = objs[sanitize_obj_name(geometry)]
-            position = Matrix(geometry.position.matrix).to_translation()
-
-            obj_child.location[0] += position[0]
-            obj_child.location[1] += position[1]
-            obj_child.location[2] += position[2]
-
-            obj_child.rotation_mode = "XYZ"
-            obj_child.rotation_euler = Matrix(geometry.position.matrix).to_euler("XYZ")
-            # this makes applying rotations correct
-            obj_child.rotation_euler[0] *= -1
-            obj_child.rotation_euler[1] *= -1
-            obj_child.rotation_euler[2] *= -1
-
-            scale = Matrix(geometry.position.matrix).to_scale()
-            obj_child.scale[0] *= scale[0]
-            obj_child.scale[1] *= scale[1]
-            obj_child.scale[2] *= scale[2]
+            geometry_mtx = Matrix(geometry.position.matrix)
+            translate = geometry_mtx.to_translation()
+            rotation = geometry_mtx.to_3x3().inverted()
+            scale = geometry_mtx.to_scale()
+            obj_child.matrix_local = Matrix.LocRotScale(translate, rotation, scale)
```

Comment:
This looks very neat. Original code took me a long time to get right, this looks much simpler.